master/buildbot/db/buildrequests.py

```diff
@@ -495,6 +495,40 @@ def thd(conn):
 
         return self.db.pool.do(thd)
 
+    def getBuildRequestsBySourcestamps(self, buildername=None, sourcestamps=None):
+        def thd(conn):
+            sourcestampsets_tbl = self.db.model.sourcestampsets
+            sourcestamps_tbl = self.db.model.sourcestamps
+            claims_tbl = self.db.model.buildrequest_claims
+            buildrequests_tbl = self.db.model.buildrequests
+            buildsets_tbl = self.db.model.buildsets
+
+            stmt = self.selectBuildSetsExactlyMatchesSourcestamps(sourcestamps=sourcestamps,
+                                                                  sourcestamps_tbl=sourcestamps_tbl,
+                                                                  sourcestampsets_tbl=sourcestampsets_tbl,
+                                                                  buildsets_tbl=buildsets_tbl)
+
+            q = sa.select([buildrequests_tbl, claims_tbl]) \
+                .where(buildrequests_tbl.c.buildsetid.in_(stmt)) \
+                .where(buildrequests_tbl.c.complete == 1) \
+                .where(buildrequests_tbl.c.results == 0) \
+                .where(buildrequests_tbl.c.buildername == buildername) \
+                .where(buildrequests_tbl.c.artifactbrid == None) \
+                .select_from(buildrequests_tbl.outerjoin(claims_tbl, (buildrequests_tbl.c.id == claims_tbl.c.brid))) \
```

Comment:
@mariangemarcano To be honest I can't remember - I think I just looked at the requirements for `_brdictFromRow` and selected what it needed. - Will remove and retest.